src/DNSC/TimeCache.hs

```diff
@@ -0,0 +1,40 @@
+module DNSC.TimeCache (
+  new,
+  none,
+  ) where
+
+import Control.Concurrent (threadDelay)
+import Data.Int (Int64)
+import Data.IORef (newIORef, readIORef, writeIORef)
+import Data.Time (defaultTimeLocale, formatTime, getCurrentTimeZone, utcToZonedTime)
+import Data.Time.Clock.System (SystemTime (..), getSystemTime, systemToUTCTime)
+
+import DNSC.Concurrent (forkLoop)
+
+
+new :: IO ((IO Int64, IO String), IO ())
+new = do
+  secRef <- newIORef 0
+  formatRef <- newIORef mempty
+
+  let step = do
+        t <- getSystemTime  {- calls clock_gettime in x86-64 linux -}
+        zt <- utcToZonedTime <$> getCurrentTimeZone <*> pure (systemToUTCTime t)
+        let seconds = systemSeconds t
+            fstring = formatTime defaultTimeLocale "%Y-%m-%d %H:%M:%S %Z" zt
```

Comment:
I guess that this line is surprisingly due to the list programming in `time`. That's why I implemented `http-date`.

Reply:
Since the log output uses String, I do not think it is efficient to use a byte-array in the current situation.
I will do the log output with a type like byte-array in another patch.

I did this because it seems to make sense to use a difference-list even at this stage.

https://github.com/khibino/dns-resolver/pull/6/commits/14a22466170eccfa42631853de9f4578946fa846